runtime/compiler/env/j9method.cpp

```diff
@@ -288,6 +288,12 @@ bool TR_ResolvedJ9Method::isMethodInValidLibrary()
    if (!strncmp(this->convertToMethod()->classNameChars(), "com/ibm/gpu/Kernel", 18))
       return true;
 
+   if (!strncmp(this->convertToMethod()->classNameChars(), "jdk/incubator/vector/FloatVector", 32))
```

Comment:
strcmp() is probably better to avoid hard-coding the length

Reply:
`strncmp()` is used because `classNameChars()` isn't necessarily NUL-terminated

Reply:
@gita-omr, do you mean to accept the exact class name only, or any class name starting with this prefix (e.g. `FloatVector$1` in addition to `FloatVector` itself)? If you want only the exact matches, then you need to also compare `classNameLength()`

Reply:
Actually, it would be safe to accept all classes in jdk/incubator/vector/. Other checks around the code are similar. Maybe let's leave like that until we need to add more classes?

Reply:
Personally I'd prefer if we could make the intention clear from the outset, because otherwise it will become hard to determine in the future. That's why I'm trying to comment on this as the case is added - it's certainly not obvious for many of the pre-existing cases here whether e.g. nested classes or other classes with a shared prefix might be matched, and whether it's important for them to match

Since all of jdk/incubator/vector/ could be accepted here, it's certainly not *necessary* to change it, but for clarity I would recommend one of the following:
- if you mean to accept just `FloatVector`, compare the length to limit it to exactly `FloatVector`; or
- if you mean to accept just `FloatVector` and its inner classes, compare the length and also separately accept anything that starts with `jdk/incubator/vector/FloatVector$`; or
- if you want to just start accepting everything in the vector package, change the prefix to `jdk/incubator/vector/`

If you really want to avoid doing any of those, could we have a comment in the code explaining that a prefix match is fine because any other classes that it accepts are also in the vector package?

Reply:
Yeah, I agree, it's easy to make it what it should be: `jdk/incubator/vector/`